pkg/internalregistry/controllers/keyid_observation_controller.go

```diff
@@ -112,6 +150,9 @@ func (c *keyIDObservation) Run(ctx context.Context, workers int) {
 	if !cache.WaitForNamedCacheSync(name, ctx.Done(), c.cacheSyncs...) {
 		return
 	}
+	if err := c.fallbackKeyIDObservation(ctx); err != nil {
+		runtime.HandleError(err)
+	}
```

Comment:
I see in the description "OCM would need to be restarted if the service account signing public key changes," but what if this one-shot attempt fails with a transient error? It sounds like it would not recover on its own.

Does it make sense to do the fallback as part of the controller sync if and only if the signing key secret does not exist?

Reply:
The controller sync is triggering on the one secret, a secret that wouldn't exisit to trigger the sync in hypershift, so I don't think I can add the fall back to the controller sync.

I can add a few retries on err.